### DISCUSSION TITLE
yq = f.Forward(q, xq): a zero order Taylor coefficient is nan

### DISCUSSION THREAD

**Discussion by Chuan701** [coin-or/CppAD#89] [General]
cppad-20160000.1 error from a known source:
yq = f.Forward(q, xq): a zero order Taylor coefficient is nan.
Corresponding independent variables vector was written to binary a file.
vector_size = 133
file_name = /tmp/fileDJXt7B
index = 62

Error detected by false result for
if( CppAD::isnan( yq[ (q+1) * index + 0 ] )
at line 224 in the file
/usr/include/cppad/local/forward.hpp
mpc: /usr/include/cppad/utility/error_handler.hpp:207: static void CppAD::ErrorHandler::Default(bool, int, const char*, const char*, const char*): Assertion `false' failed.
Aborted (core dumped)

Does everyone encountered the same problem with MPC?
Thank you!

**Comment by bradbell** [comment]
@Chuan701 What do you mean by MPC ?
I do see `mpc:` in the error message above, but when I execute the following command `git grep -i 'mpc'`, in the current master branch for CppAD, I do not get an matches.

**Reply by Chuan701** [reply]
I don't really understand what you talk.
I mean when I execute the Model predictive control, and the error would appear.

**Comment by bradbell** [comment]
I see, MPC means model predictive control. Thanks.
This error comes up a lot in many different contexts, not just Model Predictive Contorl.

For example, try running the following code which generates the message because the square root of minus one is nan:
```
# include <cppad/cppad.hpp>

int main(void)
{   bool ok = true;
    using CppAD::vector;
    using CppAD::AD;

    // f(x) = sqrt(x)
    vector< AD<double> > ax(1), ay(1);
    ax[0] = 1.0;
    CppAD::Independent(ax);
    ay[0] = sqrt(ax[0]);
    CppAD::ADFun<double> f(ax, ay);

    // evaluate f(x) at x = -1
    vector<double> x(1), y(1);
    x[0] = -1.0;
    y = f.Forward(0, x);

    return 0;
}
```

**Comment by bradbell** [comment]
Sometimes nan's are expected in some or all of the components of f.Forward(0,x). In this case you can use
https://coin-or.github.io/CppAD/doc/check_for_nan.htm
to suppress this error checking.

**Comment by bradbell** [comment]
Sometimes it is helpful to use PrintFor to detect certain error cases; e.g.,
https://coin-or.github.io/CppAD/doc/print_for_cout.cpp.htm
